DMCompiler/DM/Visitors/DMObjectBuilder.cs

```diff
@@ -124,44 +144,54 @@ class DMObjectBuilder {
                         DMASTConstantPath parentType = varOverride.Value as DMASTConstantPath;
 
                         if (parentType == null) throw new CompileErrorException(varOverride.Location, "Expected a constant path");
-                        _currentObject.Parent = DMObjectTree.GetDMObject(parentType.Value.Path);
-                        break;
+                            varObject.Parent = DMObjectTree.GetDMObject(parentType.Value.Path);
+                        return;
                     }
                     case "tag":
                         DMCompiler.Error(new CompilerError(varOverride.Location, "tag: may not be set at compile-time"));
-                        break;
```

Comment:
I actually intend on dishing out `opendream_compiletimereadonly` in a separate PR after this one. I'll leave a TODO here though, since I didn't notice that these would need to be changed.